imucapture/src/main.cpp

```diff
@@ -40,29 +41,116 @@
 #include <iomanip>
 #include <stdexcept>
 #include <string>
-#include <fstream>  //std::ofstream
+#include <fstream>
 #include <ctime>
 
 #include <cstdio>
+#include <stdlib.h>
 #include <termios.h>
 #include <unistd.h>
 #include <sys/types.h>
 #include <sys/time.h>
 
-void changemode(int dir){
+void changemode(int dir);
+int kbhit();
+
+void device_initialization(DeviceClass &device, XsPortInfo &mtPort);
+XsPortInfo scan_usb_devices();
+
+void set_message_to_packet(
+	XsDataPacket &packet,
+	XsMessage &msg,
+	XsDeviceId dev_id);
+
+inline void
+print_position_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+inline void
+print_quaternion_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+inline void
+print_euler_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+inline void
+print_timestamp_to_file(std::ofstream &outfile, XsDataPacket &packet);
+
+int main(int argc, char* argv[])
+{
+	DeviceClass device;
+	auto start = clock();
+	XsPortInfo mtPort;
```

Comment:
```suggestion
	auto mtPort = scan_usb_devices();
```
Default initialization and then overwrite it with the actual result or get the actual result from the start?